tests/integration/security/reachability/manual-mtls/reachability_test.go

```diff
@@ -0,0 +1,139 @@
+// - app A to app B using mTLS.
+// - app A to app B using mTLS-permissive.
+// - app A to app B without using mTLS.
+// In each test, the steps are:
+// - Configure authn policy.
+// - Wait for config propagation.
+// - Send HTTP/gRPC requests between apps.
+func TestReachability(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(ctx framework.TestContext) {
+
+			rctx := reachability.CreateContext(ctx, g, p)
+			systemNM := namespace.ClaimSystemNamespaceOrFail(ctx, ctx)
+
+			testCases := []reachability.TestCase{
+				{
+					ConfigFile:          "global-mtls-on.yaml",
+					Namespace:           systemNM,
+					RequiredEnvironment: environment.Kube,
+					Include: func(src echo.Instance, opts echo.CallOptions) bool {
+						// Exclude calls to the headless TCP port.
+						if opts.Target == rctx.Headless && opts.PortName == "tcp" {
+							return false
+						}
+
+						// Exclude headless->headless
+						return src != rctx.Headless || opts.Target != rctx.Headless
+					},
+					ExpectSuccess: func(src echo.Instance, opts echo.CallOptions) bool {
+						if src == rctx.Naked && opts.Target == rctx.Naked {
+							// naked->naked should always succeed.
+							return true
+						}
+
+						// If one of the two endpoints is naked, expect failure.
+						return src != rctx.Naked && opts.Target != rctx.Naked
+					},
+				},
+				{
+					ConfigFile:          "global-mtls-on-auto.yaml",
+					Namespace:           systemNM,
+					RequiredEnvironment: environment.Kube,
+					Include: func(src echo.Instance, opts echo.CallOptions) bool {
+						// Exclude calls to the headless TCP port.
+						if opts.Target == rctx.Headless && opts.PortName == "tcp" {
+							return false
+						}
+
+						// Exclude headless->headless
+						return src != rctx.Headless || opts.Target != rctx.Headless
+					},
+					ExpectSuccess: func(src echo.Instance, opts echo.CallOptions) bool {
+						if src == rctx.Naked && opts.Target == rctx.Naked {
+							// naked->naked should always succeed.
+							return true
+						}
+
+						// If one of the two endpoints is naked, expect failure.
+						return src != rctx.Naked && opts.Target != rctx.Naked
+					},
+				},
+				{
+					ConfigFile:          "global-mtls-permissive.yaml",
+					Namespace:           systemNM,
+					RequiredEnvironment: environment.Kube,
+					Include: func(src echo.Instance, opts echo.CallOptions) bool {
+						// Exclude calls to the naked app.
+						if opts.Target == rctx.Naked {
+							return false
+						}
+
+						// Exclude calls to the headless TCP port.
+						if opts.Target == rctx.Headless && opts.PortName == "tcp" {
+							return false
+						}
+						return true
+					},
+					ExpectSuccess: func(src echo.Instance, opts echo.CallOptions) bool {
+						return true
+					},
+				},
+				{
+					ConfigFile: "global-mtls-off.yaml",
+					Namespace:  systemNM,
+					Include: func(src echo.Instance, opts echo.CallOptions) bool {
```

Comment:
Now I'm reading the code, wondering is it really needed to have `Include` given we have `ExpectSuccess`? We can just customize `ExpectSuccess` to return differently?

Reply:
Maybe, may be not, I don't want to change the test behavior in this PR, yet. Personally, I think Include have different purpose/meaning. It's clearer to indicate that such test cases are not cover, in other words, they are not supported.